### DISCUSSION TITLE
Paragraph Indentation

### DISCUSSION THREAD

**Discussion by CooperHash** [LiYanan2004/MarkdownView#18] [Q&A] [answered]
there is a problem i can't handle it

> context: my data string storage in db in json format like 【- see featured playlists? \n Finally, anything you】

 then after render it work like this image

<img width="189" alt="image" src="https://github.com/LiYanan2004/MarkdownView/assets/73218815/4ad49b36-d4e6-457e-b432-4af3c42ee7f5">

i don't know why 【Finally, anything you】 this line indentation like 【· see featured playlist?】and how can i fix it

and there is another problem confused me is: how can i show multiple empty line, if my db string is json, how can i do?, because i try 【line1 test \n\n line2 test】but it don't work

**Comment by LiYanan2004** [comment]
@CooperHash For the first question: Can you provide your markdown text? Just need that part which can represent this issue.

Second question: MarkdownView full compliant CommonMark guidelines, please checkout [CommonMark Spec - Blank Line](https://spec.commonmark.org/0.30/#blank-line).  Seems like empty lines are unsupported.

**Comment by CooperHash** [comment]
@LiYanan2004  
Markdown text is:
> line1 test\n - line2 test\n\n line3 test

after i add one more \n after line2 it will work
i think it is my problem as i don't familiar with markdown format

thanks again

**Reply by LiYanan2004** [reply, marked as answer]
\n means soft break, resulting in a white space

\n\n means hard break, resulting in a new line

That's the same in lists. \n means adding a whitespace between the two. \n\n means exit this list item and open a new line.

**Reply by LiYanan2004** [reply]
You can play with [official dingus](https://spec.commonmark.org/dingus/?text=)

**Comment by CooperHash** [comment]
i want to know whether if u have social account like twitter? i am interested in your project. if u are willing to show me. 
finally thanks your reply and your project.

**Reply by LiYanan2004** [reply]
twitter：@LiYanan2004 

The same name as GitHub.😃
